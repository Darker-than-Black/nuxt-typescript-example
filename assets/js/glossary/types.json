{
  "String": "string",
  "Array": "array",
  "Number": "number",
  "Object": "object"
}
