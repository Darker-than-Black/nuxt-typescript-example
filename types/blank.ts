export interface BlankItemField {
  title: string,
  path: string,
  itemClass?: string,
  countryPath?: string
}

export interface BlankItemList extends BlankItemField {
  items: string[]
}

export interface BlankItemConfig {
  type: string,
  path: string,
  title?: string,
  countryPath?: string,
  width?: string,
  class?: string,
  headerClass?: string,
  itemsContentClass?: string,
  itemListClass?: string,
  items: BlankItemField[],
  icons?: BlankItemList[]
}
