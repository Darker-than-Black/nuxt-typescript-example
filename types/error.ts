export type HttpError = {
  field: string,
  message: string
}

export type ErrorHandler = {
  statuses: Array<number>,
  return: any
}

export type ErrorConfig = {
  returnValue?: any,
  errorsList?: Array<ErrorHandler>
}

export type ErrorCode = {
  error?: number
}
