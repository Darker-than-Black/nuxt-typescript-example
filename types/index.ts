export * from './input'
export * from './router'
export * from './error'
export * from './table'
export * from './link'
export * from './blank'

// api
export * from './api/declaration/declarations'
export * from './api/declaration/declaration'
export * from './api/car-goods'
export * from './api/entry-journal/entry-journal'
export * from './api/entry-journal/entry-journal-ticket'
export * from './api/dictionary'

// UI
export * from './UI'

// tests
export * from './unit-test'
