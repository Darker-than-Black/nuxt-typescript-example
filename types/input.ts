export type InputValue = number|string|boolean

export type MultiSelectValue = InputValue[]

export type SelectOptions = SelectOption[]

export type SelectOption = {
  value: InputValue,
  text: string
}

export type TableRowValue = {
  [key: string]: FormFieldValue
}

export type FormFieldValue = InputValue|MultiSelectValue[]|TableRowValue[]|object

export interface IFormFieldComponentProps {
  value?: FormFieldValue,
  readonly label?: string,
  readonly disabled?: boolean,
  readonly inputClasses?: string[]|string,
  // input attrs
  readonly type?: string,
  readonly placeholder?: string,
  // select attrs
  readonly options?: SelectOptions,
  readonly filterKey?: string,
  // for validate
  readonly vid?: string,
  readonly rules?: string,
  // for data list
  readonly listLink?: string,
}

export interface IFactoryFieldConfig extends IFormFieldComponentProps {
  // ~/assets/js/glossary/field-types.json
  readonly fieldType: string,
  readonly defaultValue: FormFieldValue,
  readonly invisible?: boolean,
  readonly fields?: {
    [key: string]: IFactoryFieldConfig
  }
}

export interface IFactoryFieldsConfig {
  [key: string]: IFactoryFieldConfig
}
