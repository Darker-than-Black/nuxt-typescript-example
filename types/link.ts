export type Links = {
  [key: string]: Link
}

export type Link = {
  text: string,
  icon?: string,
  path: string,
  subLinks?: Links
}
