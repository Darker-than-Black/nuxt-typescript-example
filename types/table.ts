export type TableConfig = {
  size?: number,
  totalElements?: number,
  totalPages?: number,
  number?: number // page number
}

// Doc: https://bootstrap-vue.org/docs/components/table#field-definition-reference
export type TableColConfig = {
  key: string,
  label: string,
  sortable: boolean,
  tdClass: string[] | string,
  thClass: string[] | string
}

export type TableSortEvent = {
  apiUrl: string,
  currentPage: number,
  filter: any
  perPage: number,
  sortBy: string
  sortDesc: boolean
}
