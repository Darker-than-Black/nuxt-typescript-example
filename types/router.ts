import { InputValue, MultiSelectValue } from '~/types/input'

export type QueryParam = MultiSelectValue | InputValue

export type QueryParams = {
  [key: string]: QueryParam
}

export type QueryParamsConfig = {
  [key: string]: {
    defaultValue: QueryParam,
    type: string
  }
}
