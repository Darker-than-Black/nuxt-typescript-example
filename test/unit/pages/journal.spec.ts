import VueRouter from 'vue-router'
import BootstrapVue from 'bootstrap-vue'
import VueTestInstance from '~/test/factory/vue-instance'
import EntryJournalListPage from '~/pages/journal/index.vue'
import EntryJournalListModule from '~/store/entry-journal/list'
import EntryJournalListMock from '~/test/api/entry-journal-list.json'
import { TableConfig, EntryJournalUser } from '~/types'

describe('page entry journal list', () => {
  let wrapper: any
  const router = new VueRouter()
  const serverConfig: TableConfig = {
    size: EntryJournalListMock.size,
    totalElements: EntryJournalListMock.totalElements,
    totalPages: EntryJournalListMock.totalPages,
    number: EntryJournalListMock.number
  }

  beforeEach(() => {
    wrapper = VueTestInstance(EntryJournalListPage, {
      isShallow: false,
      modules: [BootstrapVue, VueRouter],
      storeModules: {
        'entry-journal/list': {
          namespaced: true,
          state: {
            ...EntryJournalListModule.state,
            serverConfig
          },
          mutations: EntryJournalListModule.mutations,
          actions: EntryJournalListModule.actions
        }
      },
      options: { router }
    })

    wrapper.vm.debounceRoute = jest.fn()
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('test function: "getUserName"', () => {
    it('without params', () => {
      expect(wrapper.vm.getUserName()).toBe('â€”')
    })

    it('with params', () => {
      const user: EntryJournalUser = EntryJournalListMock.content[0].driver as EntryJournalUser
      expect(wrapper.vm.getUserName(user)).toBe('John Doe Tarasovych')
    })
  })

  describe('test function: "getValue"', () => {
    it('return default value', () => {
      const defaultValue: string = 'asd'
      expect(wrapper.vm.getValue({}, 'path', defaultValue)).toBe(defaultValue)
    })

    it('return search value', () => {
      const obj = { a: { b: 'asd' } }
      expect(wrapper.vm.getValue(obj, 'a.b')).toBe(obj.a.b)
    })
  })
})
