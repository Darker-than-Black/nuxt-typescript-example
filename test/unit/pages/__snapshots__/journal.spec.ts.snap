// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page entry journal list should match the snapshot 1`] = `
<div class="page container-fluid">
  <div class="row mb-3">
    <div class="col-8">
      <h1 class="title">
        Зона митного контролю
      </h1>
    </div>
    <div class="text-right col-4"><a type="button" class="btn btn-black">
        Створити талон +
      </a></div>
  </div>
  <div class="row">
    <div class="col-12">
      <table role="table" aria-busy="false" aria-colcount="6" class="table b-table table" id="__BVID__32">
        <!---->
        <!---->
        <thead role="rowgroup" class="">
          <!---->
          <tr role="row" class="">
            <th role="columnheader" scope="col" aria-colindex="1" class="table__item table__th">
              <div>Тип ат</div>
            </th>
            <th role="columnheader" scope="col" aria-colindex="2" class="table__item table__th">
              <div>Номер ат</div>
            </th>
            <th role="columnheader" scope="col" aria-colindex="3" class="table__item table__th">
              <div>ПІБ водія</div>
            </th>
            <th role="columnheader" scope="col" aria-colindex="4" class="table__item table__th">
              <div>Перевізник</div>
            </th>
            <th role="columnheader" scope="col" aria-colindex="5" class="table__item table__th"><img src="/icons/Clockface.svg" alt="Clockface" title="Час вїзду і виїзду" class="img img__size-l"></th>
            <th role="columnheader" scope="col" aria-colindex="6" aria-label="Card Number" class="table__item table__th fixed-width">
              <div></div>
            </th>
          </tr>
        </thead>
        <tbody role="rowgroup">
          <!---->
          <tr role="row" class="b-table-empty-row table__row">
            <td colspan="6" role="cell" class="">
              <div role="alert" aria-live="polite">
                <div class="text-center my-2">There are no records to show</div>
              </div>
            </td>
          </tr>
          <!---->
        </tbody>
        <!---->
      </table>
    </div>
  </div>
  <div class="row page__nav">
    <div class="d-flex align-items-center pr-0 justify-content-between col-6">
      <ul role="menubar" aria-disabled="false" aria-label="Pagination" class="pagination pagination b-pagination">
        <!---->
        <li role="presentation" aria-hidden="true" class="page-item disabled"><span role="menuitem" aria-label="Go to previous page" aria-disabled="true" class="page-link">‹</span></li>
        <!---->
        <!---->
        <li role="presentation" class="page-item active"><button role="menuitemradio" type="button" aria-label="Go to page 1" aria-checked="true" aria-posinset="1" aria-setsize="1" tabindex="0" class="page-link">1</button></li>
        <!---->
        <!---->
        <li role="presentation" aria-hidden="true" class="page-item disabled"><span role="menuitem" aria-label="Go to next page" aria-disabled="true" class="page-link">›</span></li>
        <!---->
      </ul>
      <p class="title gray small ml-2">
        Всього: 10
      </p>
    </div>
    <div class="d-flex align-items-center justify-content-end col-6">
      <p class="title gray small">
        Кількість на сторінку
      </p>
      <fieldset class="form-group m-0" id="__BVID__46">
        <!---->
        <div tabindex="-1" role="group" class="bv-no-focus-ring">
          <div role="radiogroup" tabindex="-1" class="radiobtn-group__btn bv-no-focus-ring" id="__BVID__47">
            <div class="custom-control custom-control-inline custom-radio"><input type="radio" autocomplete="off" class="custom-control-input" value="10" id="__BVID__47_BV_option_0" name="__BVID__47"><label class="custom-control-label" for="__BVID__47_BV_option_0"><span>10</span></label></div>
            <div class="custom-control custom-control-inline custom-radio"><input type="radio" autocomplete="off" class="custom-control-input" value="20" id="__BVID__47_BV_option_1" name="__BVID__47"><label class="custom-control-label" for="__BVID__47_BV_option_1"><span>20</span></label></div>
            <div class="custom-control custom-control-inline custom-radio"><input type="radio" autocomplete="off" class="custom-control-input" value="50" id="__BVID__47_BV_option_2" name="__BVID__47"><label class="custom-control-label" for="__BVID__47_BV_option_2"><span>50</span></label></div>
            <div class="custom-control custom-control-inline custom-radio"><input type="radio" autocomplete="off" class="custom-control-input" value="100" id="__BVID__47_BV_option_3" name="__BVID__47"><label class="custom-control-label" for="__BVID__47_BV_option_3"><span>100</span></label></div>
          </div>
          <!---->
          <!---->
          <!---->
        </div>
      </fieldset>
    </div>
  </div>
</div>
`;
